controller-server/src/main/java/com/yahoo/vespa/hosted/controller/rotation/RotationRepository.java

```diff
@@ -75,6 +88,57 @@ public Rotation getOrAssignRotation(Application application, RotationLock lock)
         return findAvailableRotation(application, lock);
     }
 
+    public List<AssignedRotation> getOrAssignRotations(Application application, RotationLock lock) {
+        if (application.deploymentSpec().globalServiceId().isPresent() && ! application.deploymentSpec().endpoints().isEmpty()) {
```

Comment:
`DeploymentSpec` already enforces this.

Reply:
I agree, but I think it doesn't hurt to enforce it here as well?